build.rs

```diff
@@ -125,85 +125,96 @@ fn build_c_code(out_dir: &str) -> Result<(), std::env::VarError> {
-            let program_files = env::var(program_files_env_var).unwrap();
-            let mut msbuild = std::path::PathBuf::from(&program_files);
-            msbuild.push("MSBuild");
-            msbuild.push("14.0");
-            msbuild.push("bin");
-            if let Some(amd64) = optional_amd64 {
-                msbuild.push(amd64);
-            }
-            msbuild.push("msbuild.exe");
-            let _ = try!(std::fs::metadata(&msbuild).map_err(|_| ()));
-            Ok(msbuild.into_os_string())
+    Ok(())
+}
+
+fn build_msvc(target_triple: &[&str], disable_opt: bool, num_jobs: &str,
+              lib_path: PathBuf, out_dir: &str)
+              -> Result<(), std::env::VarError> {
+    let arch = target_triple[0];
+    let (platform, optional_amd64) = match arch {
+        "i686" => ("Win32", None),
+        "x86_64" => ("x64", Some("amd64")),
+        _ => panic!("unexpected ARCH: {}", arch),
+    };
+
+    fn find_msbuild_exe(program_files_env_var: &str,
+                        optional_amd64: Option<&str>)
+                        -> Result<std::ffi::OsString, ()> {
+        let program_files = env::var(program_files_env_var).unwrap();
+        let mut msbuild = PathBuf::from(&program_files);
+        msbuild.push("MSBuild");
+        msbuild.push("14.0");
+        msbuild.push("bin");
+        if let Some(amd64) = optional_amd64 {
+            msbuild.push(amd64);
         }
+        msbuild.push("msbuild.exe");
+        let _ = try!(std::fs::metadata(&msbuild).map_err(|_| ()));
+        Ok(msbuild.into_os_string())
+    }
 
-        let msbuild = find_msbuild_exe("ProgramFiles", optional_amd64)
-            .or_else(|_| find_msbuild_exe("ProgramFiles(x86)", optional_amd64))
-            .unwrap();
+    let msbuild = find_msbuild_exe("ProgramFiles", optional_amd64)
+        .or_else(|_| find_msbuild_exe("ProgramFiles(x86)", optional_amd64))
+        .unwrap();
 
-        // .gitignore isn't packaged, so if it exists then this is not a
-        // packaged build. Otherwise, assume it is a packaged build, and use
-        // the prepackaged libs so that we don't require Perl and Yasm being
-        // installed.
-        let use_prepackaged_asm = std::fs::metadata(".gitignore").is_err();
+    // .gitignore isn't packaged, so if it exists then this is not a
+    // packaged build. Otherwise, assume it is a packaged build, and use
+    // the prepackaged libs so that we don't require Perl and Yasm being
+    // installed.
+    let use_prepackaged_asm = std::fs::metadata(".gitignore").is_err();
 
-        let configuration = if disable_opt { "Debug" } else { "Release" };
-        let args = vec![
+    let configuration = if disable_opt { "Debug" } else { "Release" };
+    let args = vec![
             format!("/m:{}", num_jobs),
             format!("/p:Platform={}", platform),
             format!("/p:Configuration={}", configuration),
             format!("/p:OutRootDir={}/", out_dir),
         ];
```

Comment:
These lines are too indented. (When I reduce the indention by 4 spaces in my copy and then run `rustfmt` it keeps it as-is.)